piker/cli.py

```diff
@@ -135,3 +144,84 @@ def watch(loglevel, broker, rate, name):
         rate = broker_limit
         log.warn(f"Limiting {brokermod.__name__} query rate to {rate}/sec")
     trio.run(_async_main, name, watchlists[name], brokermod, rate)
+
+    # broker_conf_path = os.path.join(
+    #     click.get_app_dir('piker'), 'watchlists.json')
+    # from piker.testing import _quote_streamer as brokermod
+    trio.run(_async_main, name, watchlists[name], brokermod)
+
+
+@cli.group()
+@click.option('--loglevel', '-l', default='warning', help='Logging level')
+@click.option('--config_dir', '-d', default=_watchlists_data_path,
+              help='Path to piker configuration directory')
+@click.pass_context
+def watchlists(ctx, loglevel, config_dir):
+    """Watchlists commands and operations
+    """
+    get_console_log(loglevel)  # activate console logging
+    wl.make_config_dir(_config_dir)
+    ctx.obj = {'path': config_dir,
+               'watchlist': wl.ensure_watchlists(config_dir)}
+
+
+@watchlists.command(help='show watchlist')
+@click.argument('name', nargs=1, required=False)
+@click.pass_context
+def show(ctx, name):
+    watchlist = ctx.obj['watchlist']
+    click.echo(colorize_json(
+               watchlist if name is None else watchlist[name]))
+
+
+@watchlists.command(help='load passed in watchlist')
+@click.argument('data', nargs=1, required=True)
+@click.pass_context
+def load(ctx, data):
+    try:
+        wl.write_watchlists(data, ctx.obj['path'])
+    except (json.JSONDecodeError, IndexError):
+        click.echo('You have passed an invalid text respresentation of a '
+                   'JSON object. Try again.')
+
+
+@watchlists.command(help='add ticker to watchlist')
+@click.argument('name', nargs=1, required=True)
+@click.argument('ticker_name', nargs=1, required=True)
+@click.pass_context
+def add(ctx, name, ticker_name):
+    watchlist = ctx.obj['watchlist']
+    wl.add_ticker(name, ticker_name, watchlist, ctx.obj['path'])
+
+
+@watchlists.command(help='remove ticker from watchlist')
+@click.argument('name', nargs=1, required=True)
+@click.argument('ticker_name', nargs=1, required=True)
+@click.pass_context
+def remove(ctx, name, ticker_name):
+    watchlist = ctx.obj['watchlist']
+    wl.remove_ticker(name, ticker_name, watchlist, ctx.obj['path'])
+
+
+@watchlists.command(help='delete watchlist group')
+@click.argument('name', nargs=1, required=True)
+@click.pass_context
+def delete(ctx, name):
+    watchlist = ctx.obj['watchlist']
+    wl.delete_group(name, watchlist, ctx.obj['path'])
+
+
+@watchlists.command(help='merge a watchlist from another user')
+@click.argument('watchlist_to_merge', nargs=1, required=True)
+@click.pass_context
+def merge(ctx, watchlist_to_merge):
+    watchlist = ctx.obj['watchlist']
+    wl.merge_watchlist(watchlist_to_merge, watchlist, ctx.obj['path'])
```

Comment:
as per discussion `merge_watchlist` and all the rest of these watchlist api funcs should maybe not call `write_sorted_json()` internally such that consumers of the `watchlists` module can mutate the `dict` in memory without having to write to disk if desired.